### ISSUE TITLE
Error extensions cause stack overflow

### ISSUE THREAD

**Issue by neslinesli93** [async-graphql/async-graphql#719]
## Expected Behavior
When using [error extensions](https://async-graphql.github.io/async-graphql/en/error_extensions.html), I should be able to implement the `ErrorExtensions` trait for my errors and then call `error.extend()` inside resolvers. When querying a field that returns an error, I should be able to see the error along with the `code` and other custom fields I may add

## Actual Behavior
The program crashes when calling the graphql endpoint. Here is the stacktrace

```
thread 'actix-rt|system:0|arbiter:0' has overflowed its stack
fatal runtime error: stack overflow
```

## Steps to Reproduce the Problem

1. Run the `error-extensions` example contained in the [examples folder](https://github.com/async-graphql/examples/blob/master/actix-web/error-extensions/src/main.rs#L60)
2. Call the `extend` query on the schema
3. The server crashes

Basically the code goes in a loop on [this line](https://github.com/async-graphql/async-graphql/blob/master/src/error.rs#L375), and then ultimately crashes

## Specifications

- Version: Reproduced with the branches `actix-web-v4-beta` and `master`
- Platform:
- Subsystem:

**Comment by sunli829** [comment]
Fixed in `v3.0.7` 😁
